CMake/ClangCompilers.cmake

```diff
@@ -19,7 +19,11 @@ ADD_DEFINITIONS( -Drestrict=__restrict__ )
 
 SET(CMAKE_C_FLAGS     "${CMAKE_C_FLAGS} -fomit-frame-pointer -fstrict-aliasing")
 SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fomit-frame-pointer -fstrict-aliasing -D__forceinline=inline")
-SET( HAVE_POSIX_MEMALIGN 0 )    # Clang doesn't support -malign-double
+IF (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5.0 )
```

Comment:
Shouldn't posix_memalign depend on the libc being using, not the compiler version?

That said, the old line doesn't match the comment.

Reply:
I have to say I took the scripts word for it.  It seems that in some cases clang needed changes even when posix_memalign existed on the system. However its generally something provided or not at the system level.

There is lots of discussion of posix_memalign on different plarforms in the clang 3.x time period so I image this is real and we should still set it to 0 for old enough clang.  I've fixed this so if we're on a system where we earlier failed to detect posix_memalign we don't set it to 1 here.

Reply:
If the alignement is not guaranteed. The calculation may go wrong.
I also feel this is not a compiler thing but glibc.
@PDoakORNL which platform you use requires this change?